test/Nerdbank.Streams.Tests/NilTests.cs

```diff
@@ -0,0 +1,112 @@
+using System;
+using MessagePack;
+using Xunit;
+
+namespace MessagePack.UnitTests
+{
+    /// <summary>
+    /// Unit tests for the <see cref="Nil"/> struct.
+    /// </summary>
+    public class NilTests
+    {
+        private readonly Nil _nilInstance;
```

Comment:
This `Nil` type, and some other types these new files are testing, are not actually part of Nerdbank.Streams' public API. We link it in from a git submodule and it compiles as `internal`. This test ought to not even compile. And in this case our policy is to assume the code in the submodule is already well tested in its own repo, such that we focus our testing on the code that is novel here.